src/execution/as_abi.rs

```diff
@@ -613,3 +686,88 @@ fn alloc_string_array(env: &ASEnv, vec: &[String]) -> ABIResult<i32> {
         Err(err) => abi_bail!(err),
     }
 }
+
+#[allow(dead_code)]
+/// Tooling, return a buffer (Vec<u8>) from a given offset
+fn get_buffer(memory: &Memory, ptr: i32) -> ABIResult<Vec<u8>> {
+    match BufferPtr::new(ptr as u32).read(memory) {
+        Ok(buffer) => Ok(buffer),
+        Err(err) => abi_bail!(err),
+    }
+}
+
+/// Flatten a Vec<Vec<u8>> to a Vec<u8> with the format:
+/// L (32 bits LE) V1_L (8 bits) V1 (8bits * V1_L), V2_L ... VN (8 bits * VN_L)
+fn ser_bytearray_vec(data: &Vec<Vec<u8>>, max_entry: usize) -> ABIResult<Vec<u8>> {
+    if data.len() == 0 {
+        return Ok(Vec::new());
+    }
+
+    // u16::MAX is still ok; u32::MAX -> panic!
+    if data.len() > max_entry || data.len() > u16::MAX as usize {
+        abi_bail!("Too much entry in datastore");
+    }
+
+    if !data.iter().all(|v| u8::try_from(v.len()).is_ok()) {
+        abi_bail!("Unable to serialize some keys with too much entry")
```

Comment:
`some datastore keys are too long`

Note that this is also an option:

```
let keys = vec!["key1".to_bytes().as_vec(), "key2".to_bytes().as_vec()];

let len: u16 = match keys.len().try_into() {
  Ok(v) => v,
  Err(err) => xxx_bail!("too many datastore entries")
}

let mut ser = Vec<u8> = Vec::with_capacity(4 + (len as usize * (1+255))));
ser.extend( len.to_be_bytes() )
for key in keys.into_iter() {
  let key_len: u8 = match key.len().try_into() {
    Ok(v) => v,
    Err(err) => xxx_bail!("datastore key too long")
  };
  ser.push(key_len);
  ser.extend(key);
}


```

Reply:
That's easier to read. Thanks!

Reply:
oh sorry: to_le_bytes indeed :)